site/app/models/gradeable/AutoGradedVersion.php

```diff
@@ -243,26 +243,28 @@ private function loadTestcases() {
         }
 
         // Load the testcase results (and calculate early incentive points)
-        foreach ($config->getTestcases() as $testcase) {
-            if (!isset($result_details['testcases'][$testcase->getIndex()])) {
-                // TODO: Autograding results file was incomplete.  This is a big problem, but how should
-                // TODO:   we handle this error
-            }
-            if (
-                $result_details != null
-                && count($result_details['testcases']) > $testcase->getIndex()
-                && $result_details['testcases'][$testcase->getIndex()] != null
-            ) {
-                $graded_testcase = new AutoGradedTestcase(
-                    $this->core,
-                    $testcase,
-                    $results_path,
-                    $results_public_path,
-                    $result_details['testcases'][$testcase->getIndex()]
-                );
-                $this->graded_testcases[$testcase->getIndex()] = $graded_testcase;
-                if (in_array($testcase, $config->getEarlySubmissionTestCases())) {
-                    $this->early_incentive_points += $graded_testcase->getPoints();
+        if(!is_null($config)) {
+            foreach ($config->getTestcases() as $testcase) {
+                if (!isset($result_details['testcases'][$testcase->getIndex()])) {
+                    // TODO: Autograding results file was incomplete.  This is a big problem, but how should
```

Comment:
I agree that this fix will stop the bleeding, but I also want to know why a config would be null.

Reply:
My guess is loadTestCases can be called even when there is no autograding.